data/types/bolus/combination/combination.go

```diff
@@ -61,7 +61,7 @@ func (c *Combination) Validate(validator structure.Validator) {
 	}
 
 	if c.NormalExpected != nil {
-		validator.Int("duration", c.Duration).Exists().EqualTo(DurationMinimum)
+		validator.Int("duration", c.Duration).Exists().InRange(DurationMinimum, DurationMaximum)
```

Comment:
Given our discussion, I think we need to completely separate the normal validation from the extended/duration validation. That is, I think we need to assume that a combo can have any valid normal along with any valid extended/duration. There should not be any assumption that it one is interrupted, then the other will be.

Thus, I think we need to change line 63-124 to:
```
	validator.Int("duration", c.Duration).Exists().InRange(DurationMinimum, DurationMaximum)
	durationExpectedValidator := validator.Int("expectedDuration", c.DurationExpected)
	if c.Duration != nil && *c.Duration >= DurationMinimum && *c.Duration <= DurationMaximum {
		durationExpectedValidator.InRange(*c.Duration, DurationMaximum)
	} else {
		durationExpectedValidator.InRange(DurationMinimum, DurationMaximum)
	}
	if c.ExtendedExpected != nil {
		durationExpectedValidator.Exists()
	} else {
		durationExpectedValidator.NotExists()
	}
	validator.Float64("extended", c.Extended).Exists().InRange(ExtendedMinimum, ExtendedMaximum)
	extendedExpectedValidator := validator.Float64("expectedExtended", c.ExtendedExpected)
	if c.Extended != nil && *c.Extended >= ExtendedMinimum && *c.Extended <= ExtendedMaximum {
		extendedExpectedValidator.InRange(*c.Extended, ExtendedMaximum)
	} else {
		extendedExpectedValidator.InRange(ExtendedMinimum, ExtendedMaximum)
	}

	validator.Float64("normal", c.Normal).Exists().InRange(NormalMinimum, NormalMaximum)
	normalExpectedValidator := validator.Float64("expectedNormal", c.NormalExpected)
	if c.Normal != nil && *c.Normal >= NormalMinimum && *c.Normal <= NormalMaximum {
		normalExpectedValidator.InRange(*c.Normal, NormalMaximum)
	} else {
		normalExpectedValidator.InRange(NormalMinimum, NormalMaximum)
	}
```

The above code was copied from both `extended.go` and `normal.go` with a couple small changes. Basically, to **remove** the validation that if the normal/extended is 0 then there *must* be a expectedNormal/expectedExtended present. That is not the case. (This same validation should likely be removed from `extended.go` and `normal.go`, also, since I can totally see a pump sending a 0 normal bolus without expectedNormal).

What do you think? Does the above code make sense? What about the subsequent changes to `extended.go` and `normal.go`?

Reply:
BTW, you likely can copy much of the tests from `extended_test.go` and `normal_test.go`. Assuming we make the above mentioned changes to `extended.go` and `normal.go` I'd make the changes to those test files first. Then, copy the relevant sections from those test files into `combination_test.go` to replace much of the existing related tests.

Does that make sense?